src/components/FilterBar/FilterDisplay/Inputs/RelativeDateRangeInput.react.js

```diff
@@ -0,0 +1,208 @@
+import { RangeInput } from "./RangeInput.react";
+import { DateInput } from "./DateInput.react";
+export class RelativeDateRangeInput extends RangeInput {
+  constructor(props) {
+    super(props)
+    this.state = {
+      value: this.props.value || { from: null, to: null },
+      operator: this.props.value.operator || "absolute"
+    }
+  }
+
+  isAbsolute() {
+    return this.state.operator == "absolute"
+  }
+
+  intToMoment(value) {
+    if (Number.isNaN(value) || value == null) {
+      return null
+    }
+
+    if (value < 0) {
+      return moment().subtract(Math.abs(value), 'day')
+    } else {
+      return moment().add(Math.abs(value), 'day')
+    }
+  }
+
+  momentFormatted(value) {
+    var momentValue = this.intToMoment(parseInt(value))
+
+    if (momentValue) {
+      return momentValue.format(this.props.dateFormat)
+    } else {
+      return moment().format(this.props.dateFormat)
+    }
+  }
+
+  onChangeFrom(event) {
+    var newValue = this.state.value
+    newValue["from"] = event.target.value
+    this.setState({ value: newValue })
+  }
+
+  onChangeTo(event) {
+    var newValue = this.state.value
+    newValue["to"] = event.target.value
+    this.setState({ value: newValue })
+  }
+
+  onRelativeChange(event) {
```

Comment:
Should we name this something like `onSelectChange` or `onPresetChange` (as our select is essentially a collection of presets)?
I got slightly confused when reading `onRelativeChange`, then seeing the check below `if(this.isAbsolute)`.